### ISSUE TITLE
Details grid edit

### ISSUE THREAD

**Issue by regorxxx** [TT-ReBORN/Georgia-ReBORN#91]
Ok, as said, it was easy. Now I have 2 questions, don't want to investigate all files until I get it right spending hours (for me learning the internals, and then you testing it).

1. How to force the panel to use new settings? After editing "metagrid" variable, the panel gets updated some seconds later, so I used a callback.
```
// I have used the new track callback
// But have to test it when no tracks are being played or nowPlaying returns gargabe
// Is there any other routine?
on_playback_new_track(fb.GetNowPlaying());
```
3. To save the values to the file? I see at gr-menu line 2489. Is that the right routine to save them?

```
themeConfigMenu.addItem('Save settings to config file', false, () => {
...
setThemeSettings(true) // And that's it?
});
```
![image](https://user-images.githubusercontent.com/83307074/199554104-4181e5dd-72b4-403a-b9ae-e693efe8edb7.png)
![themet](https://user-images.githubusercontent.com/83307074/199555197-77528635-cccd-47ef-8d1e-7b419a5c3140.gif)

**Comment by regorxxx** [comment]
Once I get the basics, we can go fancy with a popup showing the current rows and asking which row to edit, etc.

**Comment by TT-ReBORN** [comment]
1. It's in gr-callbacks.js -> metadata grid will be rebuild from `function on_metadb_changed(handle_list, fromhook) {`

			str.grid = [];
			for (let k = 0; k < metadataGrid.length; k++) {
				let val = $(metadataGrid[k].val);
				if (val && metadataGrid[k].label) {
					if (metadataGrid[k].age) {
						val = $('$date(' + val + ')'); // Never show time
						const age = calcAgeDateString(val);
						if (age) {
							val += ` (${age})`;
						}
					}
					str.grid.push({
						age: metadataGrid[k].age,
						label: metadataGrid[k].label,
						val
					});
				}
			}

And there is also a `function clearUIVariables() {`  which clears all, used for example on_playback_stop or in initMain()

2. No but you were right on track, the setThemeSettings() function is to save ( with parameter true ) or load ( with no parameter ) all current using SMP panel properties to the config file BUT NOT ALL are linked to the config file. 
If you open the config file ( georgia-reborn-config.jsonc ) each section has a NOTE: in the comment header that starts with:
`// * Note: These settings will`
Almost all settings are saved but some not:
"title_format_strings": {
"themeStyleCustom": {
"themeCustomFont": {
"metadataGrid": [
"settings": {
These are not saved and only for manual edits, that means user need to manually open the config file and modify them themselves. All others settings will be saved automatically to the config file via top menu Options > Settings > Theme configuration > Save settings to config file -> `setThemeSettings(true)`

The configuration file is defined via gr-configuration.js with settings from 
gr-defaults.js ( `const defaultMetadataGrid = [` ) 
and gr-settings.js ( Line 1180 ).

Hope that helps and thanks for your help!

-Tom

**Comment by regorxxx** [comment]
Uhm... 1) solved. I refactored code.

![image](https://user-images.githubusercontent.com/83307074/199592599-fc7f82fa-a219-486f-9a75-b18e87789203.png)
So on on_metadb_changed is reused with lp value.
![image](https://user-images.githubusercontent.com/83307074/199592647-b398660e-db63-46dd-82a4-0350dba68134.png)
And the same on the menu, not needing to process all again.
![image](https://user-images.githubusercontent.com/83307074/199594973-f64c31e1-1055-443b-ace0-dae7a05fde46.png)

The second took me some time, not sure why Mordred did not add an update function for arrays...
So fixed that... gr-configuration.js
![image](https://user-images.githubusercontent.com/83307074/199596579-24e4d8f2-2e36-43b6-899a-a64261d31a17.png)
And throws a popup in case you fuck up mixing config value types.
![image](https://user-images.githubusercontent.com/83307074/199596868-d6146ee3-181c-4a52-a286-6a0e7d674c55.png)

[details_grid_02_11_2022.zip](https://github.com/TT-ReBORN/Georgia-ReBORN/files/9923973/details_grid_02_11_2022.zip)

**Comment by regorxxx** [comment]
My bad, obviously parsing to JSON is not enough, will add more checks and polisishing. Was just for you to test it.

**Comment by regorxxx** [comment]
[details_grid_04_11_2022.zip](https://github.com/TT-ReBORN/Georgia-ReBORN/files/9933656/details_grid_04_11_2022.zip)
So there it is. Unless I missed something, you have to try to break it on purpose, but the same can be done with the config file so...

**Comment by TT-ReBORN** [comment]
Thanks, tested and it's working!

**Comment by TT-ReBORN** [comment]
@regorxxx,

uploaded WIP Beta 10c for us, contains all your additions.
Only some small name changes, linted your code and refactored a one liner so ESLint doesn't complain...
You should just replace all your javascript files because there were more changes I needed to do when refactored
the top menu code...

Also check latest changelog in the zip file for more information.

[Here is the zip](https://cloud.multibits.eu/s/5pesgEtCrPRbZdy).

-Tom

**Comment by regorxxx** [comment]
Why don't you use foo_openlyrics Lyrics? (just curiosity)
https://github.com/jacquesh/foo_openlyrics

**Comment by fwn0** [comment]
See this here:
https://github.com/jacquesh/foo_openlyrics/issues/177

**Comment by TT-ReBORN** [comment]
@regorxxx, 
just released[ Beta 10c](https://github.com/TT-ReBORN/Georgia-ReBORN/discussions/74#discussioncomment-4140202), wanted to let you know we will postpone this for Beta 10d and write it then in the changelog 
and make it official.

**Comment by TT-ReBORN** [comment]
Hi @regorxxx,

I am now focused on the input popup, added it to some options in the menu.
[This is here is the latest version, will be Beta 10e.](https://cloud.multibits.eu/s/PYi4RKWQn3Jn8gD)
I have made a custom popup function `const inputBox = (option) => {` in gr-menu.js.
Your metadata grid code is now also there.

Now how do we make via utils.InputBox multi input values in one popup window?
Like WilB's htmldialog ( Library > ... > Views > Configure Views ). 
I have looked up the SMP docs and there are no examples...

For example, I have made 2 popups when activating the top menu Options > Playlist > Track row > Customize track row.
 - First popup when playlist header is being displayed ( default )
 - Second popup when playlist header is deactivated ( Options > Playlist > Album header > Album header )
    
Don't tell me that we need to make a new popup framework just for that ;-)

-Tom
